### PR TITLE
[FLINK-30686][conf] Rename adaptive batch configurations

### PR DESCRIPTION
**Pull request by JunRuiLee**
## What is the purpose of the change

Simplify the configuration of adaptive batch scheduler

## Brief change log

1.Modify adaptive batch scheduler configuration default values
2.Rename the configuration of adaptive batch scheduler

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented?  not documented

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 1]
```diff
@@ -551,91 +551,6 @@ public enum SchedulerType {
                                             code(SchedulerType.AdaptiveBatch.name()))
                                     .build());
 
-    @Documentation.Section({
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:554]: `JobManagerOptions` is `@PublicEnvlving` so we should only deprecate but not remove these options.
We can exclude them from the documentation with `@Documentation.ExcludeFromDocumentation`.

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 2]
```diff
@@ -551,91 +551,6 @@ public enum SchedulerType {
                                             code(SchedulerType.AdaptiveBatch.name()))
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:551]: This config option is left behind.

**Comment by JunRuiLee** [comment]
@zhuzhurk Thanks for code review, I've addressed all comment, please take a look.

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 3]
```diff
@@ -535,10 +535,13 @@ public enum SchedulerType {
...
                             "Controls whether partitions should already be released during the job execution.");
 
+    /** @deprecated Use {@link BatchExecutionOptions#ADAPTIVE_AUTO_PARALLELISM_MIN_PARALLELISM}. */
     @Documentation.Section({
         Documentation.Sections.EXPERT_SCHEDULING,
         Documentation.Sections.ALL_JOB_MANAGER
     })
+    @Documentation.ExcludeFromDocumentation("Hidden for deprecated")
+    @Deprecated
     public static final ConfigOption<Integer> ADAPTIVE_BATCH_SCHEDULER_MIN_PARALLELISM =
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:545]: The `@Documentation.Section` should be removed.

File: flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java [hunk 4]
```diff
@@ -32,4 +37,112 @@ public class BatchExecutionOptions {
...
+    public static final ConfigOption<Integer> ADAPTIVE_AUTO_PARALLELISM_MAX_PARALLELISM =
+            key("execution.batch.adaptive.auto-parallelism.max-parallelism")
+                    .intType()
+                    .defaultValue(128)
+                    .withDeprecatedKeys("jobmanager.adaptive-batch-scheduler.max-parallelism")
+                    .withDescription(
+                            Description.builder()
+                                    .text(
+                                            "The upper bound of allowed parallelism to set adaptively if %s has been set to %s. "
+                                                    + "Currently, this option should be configured as a power of 2, "
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java:69]: The description is outdated. It was changed in a previous commit.

**Comment by JunRuiLee** [comment]
@zhuzhurk I have addressed all comments, please take a look, thanks!

File: flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java [hunk 5]
```diff
@@ -19,8 +19,13 @@
...
+import org.apache.flink.configuration.description.Description;
+
+import java.time.Duration;
 
 import static org.apache.flink.configuration.ConfigOptions.key;
+import static org.apache.flink.configuration.JobManagerOptions.SCHEDULER;
+import static org.apache.flink.configuration.description.TextElement.code;
 
 /** Configuration options for the batch job execution. */
 public class BatchExecutionOptions {
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java:31]: This class should be marked with `@PublicEvolving`

File: flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java [hunk 6]
```diff
@@ -32,4 +37,106 @@ public class BatchExecutionOptions {
...
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Duration> BLOCK_SLOW_NODE_DURATION =
+            key("execution.batch.speculative.block-slow-node-duration")
+                    .durationType()
+                    .defaultValue(Duration.ofMinutes(1))
+                    .withDeprecatedKeys(
+                            "jobmanager.adaptive-batch-scheduler.speculative.block-slow-node-duration")
+                    .withDescription(
+                            "Controls how long an detected slow node should be blocked for.");
 }
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java:142]: A private constructor should be added. A reference can be `RestartStrategyOptions`.

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 7]
```diff
@@ -602,20 +604,18 @@ public enum SchedulerType {
                                             code(SchedulerType.AdaptiveBatch.name()))
                                     .build());
 
-    @Documentation.Section({
-        Documentation.Sections.EXPERT_SCHEDULING,
-        Documentation.Sections.ALL_JOB_MANAGER
-    })
+    /** @deprecated Use {@link BatchExecutionOptions#SPECULATIVE_ENABLED}. */
+    @Documentation.ExcludeFromDocumentation("Hidden for deprecated")
+    @Deprecated
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:609]: It's better to place `@Deprecated` in the first place of annotations.

**Review by zhuzhurk — APPROVED** [review]
Thanks for addressing all the comments. @JunRuiLee 
LGTM.

**Pull request closed without merging**